### PR TITLE
added new script QRadarCreateAQLQuery

### PR DESCRIPTION
**Pull request by esharf**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/42142

## Description
the new script will allow users to build Qradar queries easier.

## Minimum version of Cortex XSOAR
- [x] 6.0.0

## Does it break backward compatibility?
- [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    to the values.
+  defaultValue: exact
+outputs:
+- contextPath: QRadarQuery
+  description: The resultant AQL query based on the inputs.
+  type: string
+subtype: python3
+dockerimage: demisto/python3:3.9.8.24399
+runas: DBotRole
+fromversion: 5.5.0
```
Comment (David-BMS) [inline Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.yml:106]: Should be from 6.0.0

Comment (David-BMS) [reply Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.yml:106]: @yaron-libman This is the script we discussed, Lets review the argument names and examples to make sure its clear for users.

File: Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    EQUAL = "{} = '{}'"
+    NOT_EQUAL = "{} != '{}'"
+    ILIKE = "{} ILIKE '%{}%'"
+    NOT_ILIKE = "{} NOT ILIKE '%{}%'"
+
+
+def fields_section(fields_list: List[str], values_list: List[str], operator: Operators = Operators.OR,
+                   match_rule: MatchRule = MatchRule.EQUAL) -> str:
+    condition_list: List[str] = []
+    for field in map(lambda x: x if ' ' not in x else f"'{x}'", fields_list):
```
Comment (yaakovi) [inline Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.py:34]: for simplicity purposes:
```suggestion
    for field in map(lambda x: f"'{x}'", fields_list):
```

**Review by yaakovi — COMMENTED** [review]
please add doc-strings to the more complex functions with output examples (as most of them are string manipulations)

File: Packs/QRadar/Scripts/QRadarCreateAQLQuery/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+| Cortex XSOAR Version | 5.5.0 |
+
+## Inputs
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| base_values_to_search | The values of the first field to search. This can be a single value or a list of comma delimited values. For example admin1,admin2 |
+| base_fields_to_search | The field names of the first field to search. This can be a single value or a list of comma delimited values. For example admin1,admin2 |
+| base_field_state | The state of the second field to search, meaning are the values in the field should be included or excluded. Valid options are include or exclude. |
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/QRadar/Scripts/QRadarCreateAQLQuery/README.md:19]: The first is called base
This is the second
This was validated with @yaron-libman

File: Packs/QRadar/Scripts/QRadarCreateAQLQuery/README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+| **Name** | **Description** |
+| --- | --- |
+| Script Type | python3 |
+| Tags | Utility |
+| Cortex XSOAR Version | 5.5.0 |
+
+## Inputs
+---
+
+| **Argument Name** | **Description** |
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/QRadar/Scripts/QRadarCreateAQLQuery/README.md:15]: The same

File: Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.py (hunk 2 above)
Comment (esharf) [reply Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.py:34]: this solution will cause problems if someone will use a function (e.g. UTF8(payload))

**Comment by ShirleyDenkberg** [comment]
@David-BMS @yaakovi Doc review completed.

File: Packs/QRadar/Scripts/QRadarCreateAQLQuery/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (yaakovi) [reply Packs/QRadar/Scripts/QRadarCreateAQLQuery/README.md]: so it should be first (argument name is `base`)

File: Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  description: The field names of the first field to search. This can be a single
+    value or a comma-separated list of values. For example admin1,admin2.
+- name: base_field_state
+  required: true
+  auto: PREDEFINED
+  predefined:
+  - include
+  - exclude
+  defaultValue: include
+  description: The state of the second field to search, meaning whether the values in
```
Comment (yaakovi) [inline Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.yml:27]:
```suggestion
  description: The state of the first field to search, meaning whether the values in
```

Comment (yaakovi) [inline Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.yml:27]:
```suggestion
  description: The state of the first field to search, meaning whether the values in
```

File: Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.yml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    the field should be included or excluded. 
+- name: base_field_match
+  required: true
+  auto: PREDEFINED
+  predefined:
+  - exact
+  - partial
+  defaultValue: exact
+  description: Whether the values of the second field should be an exact match or a partial
```
Comment (yaakovi) [inline Packs/QRadar/Scripts/QRadarCreateAQLQuery/QRadarCreateAQLQuery.yml:36]:
```suggestion
  description: Whether the values of the first field should be an exact match or a partial
```

File: Packs/QRadar/Scripts/QRadarCreateAQLQuery/README.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+| --- | --- |
+| base_values_to_search | The values of the first field to search. This can be a single value or a comma-separated list of values. For example admin1,admin2. |
+| base_fields_to_search | The field names of the first field to search. This can be a single value or a comma-separated list of values. For example admin1,admin2. |
+| base_field_state | The state of the second field to search, meaning whether the values in the field should be included or excluded. Valid options are include or exclude. |
+| base_field_match | Whether the values of the second field should be an exact match or a partial match. Valid options are exact or partial. |
```
Comment (yaakovi) [inline Packs/QRadar/Scripts/QRadarCreateAQLQuery/README.md:20]:
```suggestion
| base_field_state | The state of the first field to search, meaning whether the values in the field should be included or excluded. Valid options are include or exclude. |
| base_field_match | Whether the values of the first field should be an exact match or a partial match. Valid options are exact or partial. |
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**